modules/School Admin/studentsSettings.php

```diff
@@ -41,67 +44,43 @@
-    echo '</div>';
+    $studentGateway = $container->get(StudentGateway::class);
 
-    if ($result->rowCount() < 1) {
-        echo "<div class='error'>";
-        echo __($guid, 'There are no records to display.');
-        echo '</div>';
-    } else {
-        echo "<table cellspacing='0' style='width: 100%'>";
-        echo "<tr class='head'>";
-        echo '<th>';
-        echo __($guid, 'Name');
-        echo '</th>';
-        echo '<th>';
-        echo __($guid, 'Active');
-        echo '</th>';
-        echo '<th>';
-        echo __($guid, 'Actions');
-        echo '</th>';
-        echo '</tr>';
-
-        $count = 0;
-        $rowNum = 'odd';
-        while ($row = $result->fetch()) {
-            if ($count % 2 == 0) {
-                $rowNum = 'even';
-            } else {
-                $rowNum = 'odd';
-            }
-            ++$count;
-
-            if ($row['active'] == 'N') {
-                $rowNum = 'error';
-            }
-
-            //COLOR ROW BY STATUS!
-            echo "<tr class=$rowNum>";
-            echo '<td>';
-            echo $row['name'];
-            echo '</td>';
-            echo '<td>';
-            echo ynExpander($guid, $row['active']);
-            echo '</td>';
-            echo '<td>';
-            echo "<a href='".$_SESSION[$guid]['absoluteURL'].'/index.php?q=/modules/'.$_SESSION[$guid]['module'].'/studentsSettings_noteCategory_edit.php&gibbonStudentNoteCategoryID='.$row['gibbonStudentNoteCategoryID']."'><img title='".__($guid, 'Edit')."' src='./themes/".$_SESSION[$guid]['gibbonThemeName']."/img/config.png'/></a> ";
-            echo "<a class='thickbox' href='".$_SESSION[$guid]['absoluteURL'].'/fullscreen.php?q=/modules/'.$_SESSION[$guid]['module'].'/studentsSettings_noteCategory_delete.php&gibbonStudentNoteCategoryID='.$row['gibbonStudentNoteCategoryID']."&width=650&height=135'><img title='".__($guid, 'Delete')."' src='./themes/".$_SESSION[$guid]['gibbonThemeName']."/img/garbage.png'/></a>";
-            echo '</td>';
-            echo '</tr>';
-        }
-        echo '</table>';
-    }
+    // QUERY
+    $criteria = $studentGateway->newQueryCriteria()
+        ->sortBy(['name'])
+        ->fromArray($_POST);
+
+    $studentNoteCategories = $studentGateway->queryStudentNoteCategories($criteria);
+
+    // DATA TABLE
+    $table = DataTable::createPaginated('studentNoteCategoriesManage', $criteria);
+
+    $table->addHeaderAction('add', __('Add'))
+        ->setURL('/modules/School Admin/studentsSettings_noteCategory_add.php')
+        ->displayLabel();
+
+    $table->modifyRows(function ($values, $row) {
+        if ($values['active'] == 'N') $row->addClass('error');
+        return $row;
+    });
+    
+    $table->addColumn('name', __('Name'));
+    $table->addColumn('active', __('Active'));
```

Comment:
```suggestion
    $table->addColumn('active', __('Active'))->format(Format::using('yesNo', 'active'));
```

Reply:
Well spotted. I've done this on my feature branch to avoid conflicts.